example/advanced/server-flex-awaitable/advanced_server_flex_awaitable.cpp

```diff
@@ -0,0 +1,1188 @@
+//
+// Copyright (c) 2022 Klemens D. Morgenstern (klemens dot morgenstern at gmx dot net)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+//------------------------------------------------------------------------------
+//
+// Example: Advanced server, flex (plain + SSL)
```

Comment:
Can we please add a comment on what this advanced server does? what's _advanced_ about it? what features does this example show?

sorry for being lazy but its ~1100 lines and its a big time investment to read and then figure out if I need it.